scripts/controlnet.py

```diff
@@ -342,16 +343,32 @@ def build_sliders(module):
             def create_canvas(h, w):
                 return np.zeros(shape=(h, w, 3), dtype=np.uint8) + 255
                 
+            def onInputUpload(x):
+                if x:
+                    isSVG = x['image'].startswith("data:image/svg+xml;base64,")
+                    data = base64.b64decode(re.sub("^.*?,", "", x['image']))
+
+                    if (isSVG and svgsupport):
+                        drawing = svg2rlg(io.BytesIO(data))
+                        data = renderPM.drawToString(drawing, fmt='PNG', dpi=shared.opts.data.get("control_net_cfg_svg_dpi"))
+                        isSVG = False
+
+                    if (not isSVG):
+                        buf = io.BytesIO(data)
+                        img = Image.open(buf)
+                        return [img.width,img.height]            
+
             def svgPreprocess(inputs):
                 if (inputs):
                     if (inputs['image'].startswith("data:image/svg+xml;base64,") and svgsupport):
                         svg_data = base64.b64decode(inputs['image'].replace('data:image/svg+xml;base64,',''))
                         drawing = svg2rlg(io.BytesIO(svg_data))
-                        png_data = renderPM.drawToString(drawing, fmt='PNG')
+                        png_data = renderPM.drawToString(drawing, fmt='PNG', dpi=shared.opts.data.get("control_net_cfg_svg_dpi"))
```

Comment:
always add default value for users don't want to restart WebUI

Reply:
assume the 2nd parameter in shared.opt.get is the default value. so it is fixed